server/questions.js

```diff
@@ -96,7 +101,48 @@ exports.updateQuestionById = function(questionId, info, callback) {
 }
 
 var updateQuestionById = function(questionId, info, callback) {
-	db.updateQuestionById(questionId, info, callback);
+	updateQuestionByType(questionId,info,callback);
+}
+
+var updateQuestionByType = function(qId, infoToUpdate, callback){
+	// Get Type of question and validate it
+	db.lookupQuestionById(qId, function(err, question){
+		if(err){
+			return callback({status:500, msg:err},null);
+		}
+
+		// validate question by its type
+		var result = validateQuestionByType(infoToUpdate, question.type);
+		if (result !== null){
+			return callback({status:400, msg:result}, null)
+		}
+		db.updateQuestionById(qId, infoToUpdate, callback);
+	});
+}
+
+var validateQuestionByType = function(question, type){
```

Comment:
We need to validate the basics here, title, body, points etc.

Reply:
Yes that is a good point,